### PR TITLE
Router test cases and update url change for router

### PR DESCRIPTION
**Pull request by gysenthil**
create and update router test cases and few api changes for router
Please ignore the pom.xml of egov-location in this merge.

### CODE REVIEW

File: core/egov-workflow/src/main/java/org/egov/workflow/web/controller/RouterController.java [hunk 1]
```diff
@@ -116,7 +116,7 @@
 
 	}
 	
-	@PostMapping(value = "/_update")
+	@PostMapping(value = "/{id}/_update")
```
Comment (SumanthKP) [inline core/egov-workflow/src/main/java/org/egov/workflow/web/controller/RouterController.java]: @gysenthil  there is {id} as path parameter for this controller method where is this id used?

If id is part of url then how do we handle role based action. We cannot insert actions for each id.

File: core/egov-workflow/src/test/java/org/egov/workflow/domain/service/RouterServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    
+    public boolean checkforDuplicate(PersistRouterReq persistRouterReq){
+		PersistRouter pr = new PersistRouter();
+		pr = routerRepository.ValidateRouter(persistRouterReq);
+		System.out.println(pr);
+		if (pr != null){
+			return false;
+		}
+			else {
```
Comment (SumanthKP) [inline core/egov-workflow/src/test/java/org/egov/workflow/domain/service/RouterServiceTest.java]: @gysenthil  this code snippet is not indented using editor present in workflow.

File: core/egov-workflow/src/test/java/org/egov/workflow/persistence/repository/RouterRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    PersistRouter rt = new PersistRouter();
+    rt = getRouter();
+    pr.setRouterType(rt);
+	
+    PersistRouter persistRouter = new PersistRouter();
+    
+    routerRepository.ValidateRouter(pr);
+    persistRouter = routerRepository.ValidateRouter(pr);
+    
+    //assertTrue(rt.equals(routerRepository.ValidateRouter(pr)));
```
Comment (SumanthKP) [inline core/egov-workflow/src/test/java/org/egov/workflow/persistence/repository/RouterRepositoryTest.java]: @gysenthil  assertion is commented out here.

File: core/egov-workflow/src/test/java/org/egov/workflow/persistence/repository/RouterRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+@Test
+public void test_Should_Update_Router() {
+	
+	PersistRouterReq pr = new PersistRouterReq();
+	final RequestInfo requestInfo = new RequestInfo();
+    final User user = new User();
+    user.setId(1l);
+    requestInfo.setUserInfo(user);
+    pr.setRequestInfo(requestInfo);
+    PersistRouter rt = new PersistRouter();
```
Comment (SumanthKP) [inline core/egov-workflow/src/test/java/org/egov/workflow/persistence/repository/RouterRepositoryTest.java]: I feel creating new persistRouter might not be required here as getRouter() method is doing it.

Comment (gysenthil) [inline core/egov-workflow/src/test/java/org/egov/workflow/persistence/repository/RouterRepositoryTest.java]: removed this call from the function. please check

File: core/egov-workflow/src/test/java/org/egov/workflow/domain/service/RouterServiceTest.java (hunk 2 above)
Comment (gysenthil) [inline core/egov-workflow/src/test/java/org/egov/workflow/domain/service/RouterServiceTest.java]: Currently we are using eclipse. we will find a similar kind of indent config as you are following in the intellig and then we can fix this issue.

File: core/egov-workflow/src/main/java/org/egov/workflow/web/controller/RouterController.java (hunk 1 above)
Comment (gysenthil) [inline core/egov-workflow/src/main/java/org/egov/workflow/web/controller/RouterController.java]: @SumanthKP : Made this change and committed. please check

**Pull request merged**